File: code/difficulty.py
import sys
import pygame

FONT_NAME = "malgungothic"  # 폰트 없으면 None으로 fallback
MENU_FONT_SIZE = 25

class Difficulty:
    # [수정] name 파라미터 추가
    def __init__(self, width, height, cell, time_limit, name):
        self.width = width
        self.height = height
        self.cell = cell
        self.time_limit = time_limit
        self.name = name  # [수정] 난이도 이름 저장

# [수정] 객체 생성 시 name 지정
# (EASY: 20x20, 셀 60px, 창 1200x1200)
EASY = Difficulty(width=20, height=20, cell=60, time_limit=180, name="Easy") 
# (HARD: 30x30, 셀 50px, 창 1500x1500)
HARD = Difficulty(width=30, height=30, cell=50, time_limit=300, name="Hard") 

def _draw_button(screen, rect, text, font, color=(200,200,200), text_color=(0,0,0)):
    pygame.draw.rect(screen, color, rect)
    pygame.draw.rect(screen, (0,0,0), rect, 2)
    surf = font.render(text, True, text_color)
    screen.blit(surf, surf.get_rect(center=rect.center))

def select_difficulty():
    pygame.init()
    screen = pygame.display.set_mode((600, 400))
    pygame.display.set_caption("미로 게임 - 난이도 선택")

    try:
        font = pygame.font.SysFont(FONT_NAME, MENU_FONT_SIZE)
    except:
        font = pygame.font.SysFont(None, MENU_FONT_SIZE)

    b1 = pygame.Rect(200, 120, 200, 70)
    b2 = pygame.Rect(200, 220, 200, 70)
    clock = pygame.time.Clock()

    while True:
        screen.fill((255, 255, 255))
        _draw_button(screen, b1, "난이도 : 쉬움 (20x20)", font) 
        _draw_button(screen, b2, "난이도 : 어려움 (30x30)", font)
        pygame.display.flip()

        for e in pygame.event.get():
            if e.type == pygame.QUIT:
                pygame.quit(); sys.exit(0)
            if e.type == pygame.MOUSEBUTTONDOWN:
                if b1.collidepoint(e.pos): return EASY
                if b2.collidepoint(e.pos): return HARD
        clock.tick(60)

File: design/assets/difficulty.js
// difficulty.js

// Python Difficulty 클래스 그대로 변환
export class Difficulty {
  constructor(width, height, cell, timeLimit) {
    this.width = width;      // 미로 가로 칸 수
    this.height = height;    // 미로 세로 칸 수
    this.cell = cell;        // 셀 크기 (픽셀)
    this.time_limit = timeLimit; // 제한 시간 (초)
  }
}

// EASY 난이도
// (EASY: 20x20, 셀 60px, 창 1200x1200)
export const EASY = new Difficulty(20, 20, 60, 180);

// HARD 난이도
// (HARD: 30x30, 셀 50px, 창 1500x1500)
export const HARD = new Difficulty(30, 30, 50, 300);

/**
 * selectDifficultyFromKey(key)
 * - Python의 select_difficulty() 대신,
 *   문자열 키(easy / hard / 1 / 2 ...)로 난이도 리턴
 *
 *   예)
 *   "easy", "low", "1"  → EASY
 *   "hard", "high", "2" → HARD
 *   그 외 / undefined   → 기본 EASY
 */
export function selectDifficultyFromKey(key) {
  if (!key) return EASY;
  const k = String(key).toLowerCase();

  if (["easy", "low", "1"].includes(k)) {
    return EASY;
  }
  if (["hard", "high", "2"].includes(k)) {
    return HARD;
  }

  // 기본값
  return EASY;
}
